lib/oplog/oplogEntry.go

```diff
@@ -50,36 +50,92 @@ func (op *oplogEntry) IsRemove() bool {
 	return op.Operation == operationRemove
 }
 
+// Returns whether this is an oplog update format v2 update (new in MongoDB 5.0)
+func (op *oplogEntry) IsV2Update() bool {
+	dataVersion, ok := op.Data["$v"]
+	if !ok {
+		return false
+	}
+
+	// bson unmarshals integers into interface{} differently depending on platform,
+	// so we handle any kind of number
+	var dataVersionInt int
+	switch t := dataVersion.(type) {
+	case int:
+		dataVersionInt = t
+	case int8:
+		dataVersionInt = int(t)
+	case int16:
+		dataVersionInt = int(t)
+	case int32:
+		dataVersionInt = int(t)
+	case int64:
+		dataVersionInt = int(t)
+	case uint:
+		dataVersionInt = int(t)
+	case uint8:
+		dataVersionInt = int(t)
+	case uint16:
+		dataVersionInt = int(t)
+	case uint32:
+		dataVersionInt = int(t)
+	case uint64:
+		dataVersionInt = int(t)
+	case float32:
+		dataVersionInt = int(t)
+	case float64:
+		dataVersionInt = int(t)
+	default:
+		return false
+	}
+
+	if dataVersionInt != 2 {
+		return false
+	}
+
+	_, ok = op.Data["diff"]
+	return ok
+}
+
 // If this oplogEntry is for an insert, returns whether that insert is a
 // replacement (rather than a modification)
 func (op *oplogEntry) UpdateIsReplace() bool {
 	if _, ok := op.Data["$set"]; ok {
 		return false
 	} else if _, ok := op.Data["$unset"]; ok {
 		return false
+	} else if op.IsV2Update() {
+		// the v2 update format is only used for modifications
+		return false
 	} else {
 		return true
 	}
 }
 
 // Given an operation, returned the fields affected by that operation
 func (op *oplogEntry) ChangedFields() []string {
+	// TODO: Patch this section
```

Comment:
Any further patches required here for v2, or is this a TODO for when there is a v3 ?

Reply:
Thanks for spotting this. This shouldn't have been committed :P I'll resolve this thread.